packages/plugin-local-storage/src/LocalStorage.js

```diff
@@ -0,0 +1,267 @@
+import Storage from 'ima/storage/Storage';
+
+const TEST_KEY = 'ima-plugin-local-storage';
+
+/**
+ * Local storage class
+ */
+class LocalStorage extends Storage {
+  static get $dependencies() {
+    return ['$Window'];
+  }
+
+  /**
+   * @param {ima.window.Window} window Global window object on client
+   */
+  constructor(window) {
+    super();
+
+    /**
+     * @type {ima.window.Window}
+     */
+    this._window = window;
+
+    /**
+     * Controls whenever the options and thus the local storage is initialized.
+     *
+     * @type {boolean}
+     * @private
+     */
+    this._initialized = false;
+
+    /**
+     * Holds current storage options.
+     *
+     * @type {Object}
+     * @private
+     */
+    this._options = {};
+  }
+
+  /**
+   * Initialize the local storage.
+   *
+   * @override
+   * @param {{
+   *   expires: ?(number|Date)=,
+   * }=} options Local storage options, you can define default expiration time
+   * @returns {LocalStorage} This local storage.
+   */
+  init(options = {}) {
+    this._options = Object.assign({}, options) || {};
+    this._initialized = true;
+
+    return this;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  has(key) {
+    return !!this.get(key);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  get(key) {
+    if (!this._initialized) {
+      return undefined;
+    }
+
+    let rawItem = localStorage.getItem(key);
+    let item;
+
+    try {
+      item = JSON.parse(rawItem);
+    } catch (error) {
+      item = rawItem;
+    }
+
+    if (item && item.value) {
+      if (item.expires && Date.now() >= item.expires) {
+        this.delete(key);
+
+        return undefined;
+      } else {
+        return item.value;
+      }
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Sets the local storage entry identified by the specified key to the provided
+   * value along with information about it's expiration, if provided. The value is
+   * run through JSON.stringify, before saving.
+   *
+   * @override
+   * @param {string} key The key identifying the local storage entity
+   * @param {*} value The local storage entry value
+   * @param {{
+   *   expires: ?(number|Date)=,
+   * }=} options Local storage options, overwrites global settings
+   * @returns {LocalStorage} This local storage
+   */
+  set(key, value, options = {}) {
+    if (!this._initialized) {
+      return this;
+    }
+
+    let opts = Object.assign({}, this._options, options);
+
+    if (value === undefined) {
+      this.delete(key);
+    } else {
+      let item = { value };
+      let expires = this._getExpires(opts);
+
+      if (expires) {
+        item.expires = expires;
+      }
+
+      localStorage.setItem(key, JSON.stringify(value));
```

Comment:
What happens if localStorage is not supported or calling setItem throws Error.

Reply:
The local storage now checks the support during an `init()` method call. So this situation should now never happen, as local storage doesn't initialize unless it complies with all the requirements. I've also wrapped the method in try/catch block. so in case of a situation, where user for example runs out of space in the middle of application run, it will not shutdown whole application.